src/main/java/org/folio/service/file/SplitFileProcessingService.java

```diff
@@ -0,0 +1,133 @@
+import org.folio.rest.jaxrs.model.InitJobExecutionsRqDto;
+import org.folio.rest.jaxrs.model.JobExecution;
+import org.folio.rest.jaxrs.model.JobExecutionDto;
+import org.folio.rest.jaxrs.model.JobExecutionDtoCollection;
+import org.folio.rest.jaxrs.model.UploadDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service containing methods to manage the lifecycle and initiate processing of
+ * split files.
+ */
+@Service
+public class SplitFileProcessingService {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private DataImportQueueItemDao queueItemDao;
+
+  @Autowired
+  public SplitFileProcessingService(DataImportQueueItemDao queueItemDao) {
+    this.queueItemDao = queueItemDao;
+  }
+
+  /**
+   * Registers split parts as Job Executions in mod-source-record-manager
+   * and adds each part to the DI queue.
+   *
+   * @return a {@link CompositeFuture} of {@link JobExecutionDto}
+   */
+  public CompositeFuture registerSplitFiles(
+    UploadDefinition parentUploadDefinition,
+    JobExecution parentJobExecution,
+    ChangeManagerClient client,
+    int parentJobSize,
+    String tenant,
+    List<String> keys
+  ) {
+    List<Future<JobExecutionDto>> futures = new ArrayList<>();
+
+    int partNumber = 1;
+    for (String key : keys) {
+      InitJobExecutionsRqDto initJobExecutionsRqDto = new InitJobExecutionsRqDto()
+        .withFiles(Arrays.asList(new File().withName(key)))
+        .withParentJobId(parentJobExecution.getId())
+        .withJobPartNumber(partNumber)
+        .withTotalJobParts(keys.size())
+        .withSourceType(InitJobExecutionsRqDto.SourceType.COMPOSITE)
+        .withUserId(
+          Objects.nonNull(parentUploadDefinition.getMetadata())
+            ? parentUploadDefinition.getMetadata().getCreatedByUserId()
+            : null
+        );
+
+      // outer scope variable could change before lambda execution, so we make it final here
+      final int thisPartNumber = partNumber;
+
+      Promise<JobExecutionDto> promise = Promise.promise();
+      futures.add(promise.future());
+
+      client.postChangeManagerJobExecutions(
+        initJobExecutionsRqDto,
+        response -> {
+          try {
+            if (
+              response.result().statusCode() != HttpStatus.HTTP_CREATED.toInt()
+            ) {
+              LOGGER.warn(
+                "registerSplitFiles:: Error creating new child JobExecution for key {}. Status message: {}",
+                key,
+                response.result().statusMessage()
+              );
+              throw new HttpException(
+                response.result().statusCode(),
+                "Error creating new JobExecution"
+              );
+            } else {
+              BufferMapper
+                .mapBufferContentToEntity(
+                  response.result().bodyAsBuffer(),
+                  JobExecutionDtoCollection.class
```

Comment:
```suggestion
                  InitJobExecutionsRsDto.class
```
I've found that this endpoint seems to return a response that corresponds to another json schema https://github.com/folio-org/mod-source-record-manager/blob/9dc47fd905ce2f90bd7368dc15e26d7180a1287c/ramls/change-manager.raml#L67 and we should use InitJobExecutionsRsDto class for response deserialization to avoid error during deserialization

Reply:
good catch, thanks!